gen-v2/stl/OperationalCredentialCluster.cpp

```diff
@@ -0,0 +1,187 @@
+                else if (tag == TLV::ContextTag(kFieldIdAdminVendorId)) {
+                    ReturnErrorOnFailure(reader.Get(adminVendorId));
+                }
+            }
+
+            ReturnErrorOnFailure(reader.ExitContainer(outer));
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    namespace OpCertResponse {
+        CHIP_ERROR Type::Encode(TLV::TLVWriter &writer, uint64_t tag) {
+            TLV::TLVType outer;
+            ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outer));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kFieldIdStatusCode), statusCode));
+            ReturnErrorOnFailure(writer.Put(TLV::ContextTag(kFieldIdFabricIndex), fabricIndex));
+            ReturnErrorOnFailure(writer.PutString(TLV::ContextTag(kFieldIdDebugText), debugText.c_str(), debugText.length()));
+            ReturnErrorOnFailure(writer.EndContainer(outer));
+            ReturnErrorOnFailure(writer.Finalize());
+            return CHIP_NO_ERROR;
+        }
+
+        CHIP_ERROR Type::Decode(TLV::TLVReader &reader) {
+            CHIP_ERROR err;
+            TLV::TLVType outer;
+
+            err = reader.EnterContainer(outer);
+            ReturnErrorOnFailure(err);
+
+            while ((err = reader.Next()) == CHIP_NO_ERROR) {
+                uint64_t tag = reader.GetTag();
+
+
+                if (tag == TLV::ContextTag(kFieldIdStatusCode)) {
+                    ReturnErrorOnFailure(reader.Get(statusCode));
+                }
+                else if (tag == TLV::ContextTag(kFieldIdFabricIndex)) {
+                    ReturnErrorOnFailure(reader.Get(fabricIndex));
+                }
+                else if (tag == TLV::ContextTag(kFieldIdDebugText)) {
+                   int len = reader.GetLength() + 1;
+                   char tmp[len];
+                   ReturnErrorOnFailure(reader.GetString(tmp, len + 1));
+                   debugText = std::string(tmp, len + 1);
+                   debugText.resize(len - 1);
+                }
+            }
+
+            ReturnErrorOnFailure(reader.ExitContainer(outer));
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    namespace OpCsrRequest {
+        CHIP_ERROR Type::Encode(TLV::TLVWriter &writer, uint64_t tag) {
+            TLV::TLVType outer;
+            ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outer));
+            ReturnErrorOnFailure(writer.PutBytes(TLV::ContextTag(kFieldIdCsrNonce), &csrNonce[0], csrNonce.size()));
+            ReturnErrorOnFailure(writer.EndContainer(outer));
+            ReturnErrorOnFailure(writer.Finalize());
+            return CHIP_NO_ERROR;
+        }
+
+        CHIP_ERROR Type::Decode(TLV::TLVReader &reader) {
+            CHIP_ERROR err;
+            TLV::TLVType outer;
+
+            err = reader.EnterContainer(outer);
+            ReturnErrorOnFailure(err);
+
+            while ((err = reader.Next()) == CHIP_NO_ERROR) {
+                uint64_t tag = reader.GetTag();
```

Comment:
We should avoid such checks, since that makes it less 'forward compatible' to changes past v1 where non-context tags are used (e.g we're entertaining the use of protocol-specific tags for manufacturer extensions). In those cases, adding too much validation will prevent that compatibility from happening.